json-fleece-aeson/src/Fleece/Aeson/Decoder.hs

```diff
@@ -25,21 +27,26 @@ import qualified Fleece.Core as FC
 data Decoder a
   = Decoder FC.Name (Aeson.Value -> AesonTypes.Parser a)
 
-fromValue :: Decoder a -> Aeson.Value -> Either String a
-fromValue =
-  AesonTypes.parseEither . toParser
+fromValue :: Decoder a -> Aeson.Value -> Either (String, String) a
+fromValue decoder value =
+  mapLeft (\(path, err) -> (AesonTypes.formatError path err, err)) $
+    AesonTypes.iparseEither (toParser decoder) value
 
 toParser :: Decoder a -> Aeson.Value -> AesonTypes.Parser a
 toParser (Decoder _name f) =
   f
 
 decode :: Decoder a -> LBS.ByteString -> Either String a
 decode decoder =
-  fromValue decoder <=< Aeson.eitherDecode
+  mapLeft fst . fromValue decoder <=< Aeson.eitherDecode
+
+decodePretty :: Decoder a -> LBS.ByteString -> Either String a
+decodePretty decoder =
+  mapLeft snd . fromValue decoder <=< Aeson.eitherDecode
```

Comment:
Rather that offer a separate `decodePretty` function it would be better to make the `decode` function return a new error type other than `String` that the caller could then decide how to format.